### PR TITLE
Add script report and print format for Assessment Result

### PR DESCRIPTION
**Pull request by crossxcell99**
[Resolved merge conflicts from #9148 ]

Added the following fields to some of the doctypes:
In Assessment Result: Link for Academic term, Academic year and course, so the results can be filtered in the report (script report name = ResultsReport).
In Assessment Plan: Check box for multiple courses, a child table (doctype: Courses Table) and a button for the getting the courses automatically from program linked in student group (if possible).

Used a workaround for getting column names and removing extra fields ("_id" and "id") for the print format (resultsreport.html)

resultsreport1
This report lists all the student results including the assessment criterias

resultprint
This is the print format of the results report

**Note**: This PR is an update on the suggestion I made in #8737

### CODE REVIEW

**Comment by manassolanki** [comment]
Hi @crossxcell99 ,
Please squash your commits if possible and rebase your branch from upstream develop.
You don't need to merge develop branch into your branch.

**Comment by crossxcell99** [comment]
Commits have been squashed into one and branch rebased from upstream develop 
@manassolanki

File: erpnext/schools/api.py [hunk 1]
```diff
@@ -289,3 +291,48 @@ def update_email_group(doctype, name):
...
+		result_name = i.name
+		results_h = frappe.get_doc('Assessment Result',i)
+		course = frappe.get_doc("Assessment Plan",results_h.assessment_plan).course
+		total_score = results_h.total_sco
+		maximum_score = results_h.maximum_score
+		grade = results_h.grade
+		score_breakdown = []
+		for j in results_h.details:
+			score_breakdown.append({'criteria':j.assessment_criteria,'maximum_score':j.maximum_score,'score':j.score})
+		results_obj.append({
```
Comment (mbauskar) [inline erpnext/schools/api.py]: indentation?

File: erpnext/schools/doctype/assessment_plan/assessment_plan.js [hunk 2]
```diff
@@ -51,6 +56,31 @@ frappe.ui.form.on("Assessment Plan", {
...
+
+    get_courses: function(frm) {
+        console.log("######   CALLED    #########");
+        frm.set_value("courses_table",[]);
+        frappe.call({
+            method: "erpnext.schools.api.get_courses",
+            args: {"student_group": frm.doc.student_group},
+            callback: function(r) {
+                console.log("######   CALLBACK RESULT    #########");
+                console.log(r.message);
```
Comment (mbauskar) [inline erpnext/schools/doctype/assessment_plan/assessment_plan.js]: remove console.log

**Review by mbauskar — CHANGES_REQUESTED** [review]
Also please remove the unnecessary whitespaces

**Comment by mbauskar** [comment]
@crossxcell99,

Thanks for the PR, Please check and fix the Codacy issues, Closing this PR for now
Push the changes and reopen the PR

**Pull request closed without merging**